packages/foundry/contracts/test/oracles/CrabOracleTest.t.sol

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity >=0.8.13;
+
+import "forge-std/src/Test.sol";
+import "@yield-protocol/utils-v2/contracts/access/AccessControl.sol";
+import "../../oracles/crab/CrabOracle.sol";
+import {ICrabStrategy} from "../../oracles/crab/CrabOracle.sol";
+import "../../oracles/uniswap/UniswapV3Oracle.sol";
+
+contract CrabOracleTest is Test {
+    CrabOracle public crabOracle;
+    bytes6 baseId = 0x323900000000;
+    bytes6 quoteId = 0x303000000000;
+    UniswapV3Oracle uniswapV3Oracle =
+        UniswapV3Oracle(0x358538ea4F52Ac15C551f88C701696f6d9b38F3C);
+
+    function setUp() public {
+        crabOracle = new CrabOracle();
+        crabOracle.grantRole(0xac426579, address(this));
+        crabOracle.setSource(
+            baseId,
+            0x313900000000,
+            ICrabStrategy(0x3B960E47784150F5a63777201ee2B15253D713e8),
+            IOracle(address(uniswapV3Oracle))
+        );
+
+        vm.startPrank(0x3b870db67a45611CF4723d44487EAF398fAc51E3);
+        uniswapV3Oracle.grantRole(
+            0xe4650418,
+            0x3b870db67a45611CF4723d44487EAF398fAc51E3
+        );
```

Comment:
This is fine though I usually prefer making a state variable of the address at the top so someone reading the test knows exactly what it represents. So in this case something like `address timelock = <that_address>`. I also call the function selector for the same reason instead of hard coding it.